### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by toboid**
#79

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -10,19 +9,20 @@ var Version = require('./version');
 var internals = {};
```
Comment (chyld) [inline lib/index.js]: you're not using internals

File: lib/start.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Load modules
+
+var Hoek = require('hoek');
+var Index = require('./');
+
+
+// Declare internals
+
+var internals = {};
```
Comment (chyld) [inline lib/start.js]: not using internals

File: lib/start.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Load modules
```
Comment (chyld) [inline lib/start.js]: start.js should be in the root directory

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+describe('Server', function () {
+
+    it('listens on supplied port', function (done) {
+
+        Index.init(8000, function (err, server) {
+
+            expect(err).to.not.exist();
+            expect(server.info.port).to.equal(8000);
+            server.stop();
```
Comment (chyld) [inline test/index.js]: put done inside stop ... server.stop(done)

File: test/index.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            var routes = server.connections[0].table();
+            var foundVersion = false;
+            routes.forEach(function (route) {
+                if (route.path === '/version') {
+                    foundVersion = true;
+                }
+            });
+            expect(foundVersion, 'version route not found').to.equal(true);
+            server.stop();
+            done();
```
Comment (chyld) [inline test/index.js]: done inside stop

Comment (toboid) [inline test/index.js]: Thanks for feedback, I won't have access a computer for a few days now so won't be able to update this. I appreciate the comments though

File: lib/index.js (hunk 1 above)
Comment (toboid) [inline lib/index.js]: Yes, I wondered about this and not really comfortable with having a redundant variable there either. However I think this is required by the [style guide](http://hapijs.com/styleguide):
> internals - must be declared as an object at the top of each module immediate following the require section

Also had a quick look at the hapi source code and found examples of the empty variable that is not used in the rest of the file there too e.g. https://github.com/hapijs/hapi/blob/master/lib/index.js.

Be interested to get @hueniverse take on this.

File: lib/start.js (hunk 3 above)
Comment (toboid) [inline lib/start.js]: Seemed to be different views on this as it wasn't specified in the task, but updated as think makes sense in the root directory

**Comment by EvanMarkHopkins** [comment]
I think it was actually decided that ```start.js``` would go in lib. I will look for the PR

File: lib/index.js [hunk 6]
```diff
@@ -10,19 +9,20 @@ var Version = require('./version');
...
 
 
-internals.init = function () {
+exports.init = function (port, callback) {
```
Comment (EvanMarkHopkins) [inline lib/index.js]: use ```next``` instead of ```callback```, also in #80 (i think)

**Comment by zoe-1** [comment]
@EvanMarkHopkins  is right. start.js goes in lib directory.

**Comment by zoe-1** [comment]
Wow! @EvanMarkHopkins is going beyond the call of duty.  He is giving references to 
PRs for each recommendation.  nice work @EvanMarkHopkins. very kind of you.

**Comment by EvanMarkHopkins** [comment]
@zoe-1 I'm asking them to revert something they just finished *fixing*, the least I can do is validate my claims :smile:

**Comment by zoe-1** [comment]
@EvanMarkHopkins  just wanted to give you a compliment. :-) 
Thank you for doing diligent work. Do not often see people making the effort like that
so I was impressed.

**Comment by toboid** [comment]
thanks for the link, missed that. moving back to lib

File: lib/index.js (hunk 6 above)
Comment (toboid) [inline lib/index.js]: thanks :smile:

**Pull request closed without merging**